e2e-tests/src/driver.rs

```diff
@@ -170,6 +176,10 @@ impl Driver {
 
             let mut applications = read_dir(&protocol_path).await?;
             while let Some(app) = applications.next_entry().await? {
+                let protocol_name = protocol_name.to_string().to_lowercase();
```

Comment:
you shouldn't need to make a string here, pach `TestContext` to take `Protocol` instead of `&str`